nmsg/kafkaio.c

```diff
@@ -224,6 +229,66 @@ _kafka_config_set_option(rd_kafka_conf_t *config, const char *option, const char
 	return true;
 }
 
+static bool
+_kafka_process_config_apply(kafka_ctx_t ctx, rd_kafka_conf_t *config, const struct my_config_item *items)
+{
+	do {
+		const char *key = my_config_item_key(items);
+		const char *value = my_config_item_value(items);
+		if (!_kafka_config_set_option(config, key, value)) {
+			_nmsg_dprintf(3, "%s: failed to set kafka configuration value %s to %s.\n",__func__, key, value);
+			return false;
+		}
+		_nmsg_dprintf(3, "%s: set kafka configuration value %s to %s.\n",__func__, key, value);
+		items = my_config_next_item(items);
+	} while(items != NULL);
+
+	return true;
+}
+
+static bool
+_kafka_process_config(kafka_ctx_t ctx, rd_kafka_conf_t *config)
```

Comment:
Returning numerical zero on success is convention but returning `false` on success is a bit strange...